### PR TITLE
Scrollbar [ready for review and merge]

### PR DESCRIPTION
**Pull request by mattkrick**
**In this PR**:
- [x] Moves more components to `react-emotion`
- [x] Introduces a `SexyScrollbar` component that needs some more TLC and is not being used yet
- [x] Retains current overflow implementation for team navigation, task columns, and agenda list

**Next steps here**:
- [ ] merge this effort
- [x] create next-steps issue #2198

**Tests** (rewritten as AC on #2198)
- create a bunch of teams & shorten the height of your window
- [x] when hovering over the teams, a sexy scrollbar appears. when leaving the sidebar, the scrollbar goes away.
- [ ] in a dashboard with cards in many columns, hovering over each column makes the scrollbar appear, showing your current location within the column.
- [ ] dragging a card in a column works as expected & there is no extra scrollbar that appears on the side
- [ ] the same scrollbar works in the agenda list, and the agenda list does not scroll until there is not enough room

### CODE REVIEW

**Comment by ackernaut** [comment]
reviewing this now

**Comment by ackernaut** [comment]
### Test No. 1 (TA)
- create a bunch of teams & shorten the height of your window
- [x] when hovering over the teams, a sexy scrollbar appears. when leaving the sidebar, the scrollbar goes away.

**Comment by mattkrick** [comment]
wow, yours looks nothing like it should!
the sidebar scrolling is broken on firefox (the window just grows as the # of teams grows) so I'm guessing you're on chrome OSX?

scrollbar

**Comment by ackernaut** [comment]
Yes, @mattkrick I was on Chrome/OSX. I noticed in your screen capture that the scrollbar fades in once you hover, which is nice, and I think is the minimal indication necessary to know there is more content that is hidden away.

I can test again locally, or kick the tires on the dev env.

**Comment by mattkrick** [comment]
yeah, the position of the bar is key! personally, i'm not a fan of the shadows because if you're in the middle of the list, it looks like the list is sunk into the sidebar, which isn't great (the agenda items is an easy example).

would you feel comfortable troubleshooting the styles? it may be awhile before i get my macbook back. i assume it has to do with `renderThumbVertical`, maybe start by removing that, & if it works try doing styles instead of a class, or including a few more props, etc.

**Comment by ackernaut** [comment]
I’ll give it a look re: scrollbar style troubles

**Comment by mattkrick** [comment]
confirmed working on chrome on linux & windows
stuck on fixing the bug in macOS.
@jordanh can you confirm the bug is in macOS? this is on the dev server right now. the problem is likely line 37 in SexyScrollbar.js if you'd like to take a stab at it.

**Comment by mattkrick** [comment]
found an article that showed how to override the macOS settings by doing this:
```
::-webkit-scrollbar {
    width:9px;
}

::-webkit-scrollbar-track {
    -webkit-border-radius:5px;
    border-radius:5px;
    background:rgba(0,0,0,0.1);
}

::-webkit-scrollbar-thumb {
    -webkit-border-radius:5px;
    border-radius:5px;
    background:rgba(0,0,0,0.2);
}

::-webkit-scrollbar-thumb:hover {
    background:rgba(0,0,0,0.4);
}

::-webkit-scrollbar-thumb:window-inactive {
    background:rgba(0,0,0,0.05);
}
```
http://blog.0100.tv/2012/11/webkit-scrollbars-on-os-x/

the only catch is that this must be applied globally & synchronously, which means it's an all-or-nothing solution. my guess is apple got lazy & only looks for a global css selector & if it finds one, it turns off native OS behavior. so, the above code overrides the native behavior, and then the SexyScrollbar overrides the above behavior. That means we'll want to apply SexyScrollbar to everything that scrolls, otherwise there will be an omni-present scrollbar.

**Comment by ackernaut** [comment]
Hey @mattkrick just took a look at this locally and it’s working for the sidebar! I was wondering, do we intend to use this wherever we expect there to be overflow, like the task columns? Also, it is possible to set the orientation (say we want to show a horizontal scrollbar because we enforce min-width for task columns and it causes overflow)?

**Comment by mattkrick** [comment]
yep, the reason we had to stick with this package is because it supports horizontal scrolling, it's the only one that really did it.

i would love to do this for task columns, but i noticed that it didn't work easily when i tried to plug it in (everything looked great until i went to drag a card, then a 2nd scrollbar popped up). it's basically just a big game of toggling `overflow` and `position` for all the parent divs until i hit the winning combination. 

I'd love for you to take a look & see if you could fix it up for the task columns & agenda list! i figure after those 2 things the PR will be ready to merge.

**Comment by mattkrick** [comment]
@ackernaut i updated the acceptance criteria. the bottom 2 checkboxes are where i got frustrated with the CSS.

**Comment by ackernaut** [comment]
@mattkrick thanks! Re: horizontal scrolling, I’m glad as I think we can use it soon. Also, I’ll take a look at those other column components and see if I can get it working. Thanks for pressing ahead on this one!!

**Comment by ackernaut** [comment]
I’ve been banging on this so I wanted to give a status. I pushed a commit and…

The yay news:
- We get `TaskColumn` in emotion
- The custom scrollbars work on the task columns, but with the bad news below
- It’s easy to target the nested `div` added by `react-custom-scrollbars` and set `visibility: hidden` on the duplicate scrollbar

The sad news:
- The scrollbars get added, and there is a false bit of overflow, on all components using the custom scrollbars. Even if there is enough room for task cards, nav items, or agenda topics, there is a slight bit of overflow caused by how `react-custom-scrollbars` renders the inner divs and sets up the height. I’ve have tried overriding properties using emotion and `& > div` syntax, which does apply the overrides correctly, but doesn’t solve the problem. I have even tried overriding the height of the inner `div` to correct the problem. 999 ways, but I’m only on like 27. This is a blocker IMO. I’ve looked over the source code for `react-custom-scrollbars` but may have to dig more. *Possible solution*: I’m wondering if we can check height before we apply the custom scrollbar, only applying when the inner height is greater than the outer height of our own containers. If greater, wrap the inner components in the custom scrollbar component, otherwise return the inner components.
- Seems we lose the effect of `react-dnd-scrollzone` on the task card columns. The custom scrollbar is at play alongside the scroll zone component. I haven’t dug into this one as much. Probably should go look at the source for `react-dnd-scrollzone`. We may just need to roll a magical package that does both custom scrollbars and configurable scroll zones in one. One scroll manager to rule them all. I can try some different configuration and/or nesting here to see if there is any different.

**Comment by mattkrick** [comment]
i think you're right, we'll need to tweak react-dnd-scrollzone to support our usecase, i can take that on.

did you manage to eliminate the extra scroll on the agenda list? if you can solve that, that same strategy can probably be applied to task columns. a good place to start would be comparing it to the team sidebar. both components have a list & then a button that sits below them. not sure why it works for the team sidebar but not the agenda list. if you can figure that out, we win.

**Comment by ackernaut** [comment]
@mattkrick the team sidebar and the agenda list have the same unwanted behavior for me on the Mac. If there is plenty of room there should be no scrollbar on either list, however there is a slight false overflow on both lists because of how the package handles the inner div with negative margins, attempting to hide the scrollbars. I think if we check heights before applying the scrollbar we may solve this one too.

We have our container and our inner content. The inner content sometimes is greater than our container. Let’s only wrap the inner content with the custom scrollbars component when the inner content is greater than our container.

**Comment by ackernaut** [comment]
Also, I may need to set up some inner gutters so that box shadows don’t get clipped by the custom scrollbar component which relies on a parent `dive` with `overflow: hidden`. This happens with the task card columns, clipping the shadow of the cards.

File: src/universal/styles/theme/globalStyles.js [hunk 1]
```diff
@@ -81,6 +81,24 @@ export default `
     font-weight: 600;
   }
 
+  /*::-webkit-scrollbar {
```
[the remark this answers is not in the document]
Comment (ackernaut) [reply src/universal/styles/theme/globalStyles.js]: You want me to nuke the webkit scrollbar declarations and then merge?

[the remark this answers is not in the document]
Comment (mattkrick) [reply src/universal/styles/theme/globalStyles.js]: oh doh, nevermind, i see that they are commented out! that's good nuff for me, merging

**Pull request merged**